jumpscale/packages/threebot_deployer/chats/threebot.py

```diff
@@ -108,6 +110,60 @@ def _verify_password(self, password):
         pubkey = NACL(seed).get_verify_key_hex()
         return pubkey == user.pubkey
 
+    def _select_node(self):
+        if self.node_policy == "Specific node":
+            return self.selected_node
+        else:
+            return super()._select_node()
+
+    @chatflow_step(title="Deployment location policy")
+    def choose_location(self):
+        self._get_available_farms()
+        self.farms_by_continent = deployer.group_farms_by_continent(self.available_farms)
+        choices = ["Automatic", "Farm", "Specific node"]
+        if self.farms_by_continent:
+            choices.insert(1, "Continent")
+        self.node_policy = self.single_choice(
+            "Please select the deployment location policy.", choices, required=True, default="Automatic"
+        )
+
+    def _ask_for_continent(self):
+        continent = self.single_choice(
+            "Please select the continent you would like to deploy your 3Bot in.",
+            list(self.farms_by_continent.keys()),
+            required=True,
+        )
+        self.available_farms = self.farms_by_continent[continent]
+
+    def _ask_for_farm(self):
+        farm_name_dict = {farm.name: farm for farm in self.available_farms}
+        farm_name = self.single_choice(
+            "Please select the farm you would like to deploy your 3Bot in.", list(farm_name_dict.keys()), required=True
+        )
+        self.available_farms = [farm_name_dict[farm_name]]
+
+    def _ask_for_node(self):
+        nodes = deployer.get_all_farms_nodes(self.available_farms, **self.query)
+        node_id_dict = {node.node_id: node for node in nodes}
+        node_id = self.single_choice(
+            "Please select the node you would like to deploy your 3Bot on.", list(node_id_dict.keys()), required=True
+        )
+        self.selected_node = node_id_dict[node_id]
+        self.available_farms = [farm for farm in self.available_farms if farm.id == self.selected_node.farm_id]
+
+    @chatflow_step(title="Deployment location")
+    def choose_deployment_location(self):
+        if self.node_policy == "Automatic":
+            pass
```

Comment:
if it's pass, then we can remove this block unless we need it for something later ?